rotkehlchen/data_import/importers/rotki_trades.py

```diff
@@ -37,30 +34,39 @@ def _consume_rotki_trades(
         - DeserializationError
         - UnknownAsset
         - KeyError
-        - DivisionByZero if `amount_bought` is ZERO
         """
         amount_sold = deserialize_asset_amount(csv_row['Sell Amount'])
         amount_bought = deserialize_asset_amount(csv_row['Buy Amount'])
-        asset, fee, fee_currency, location, timestamp = process_rotki_generic_import_csv_fields(csv_row, 'Base Currency')  # noqa: E501
-        if (trade_type := TradeType.deserialize(csv_row['Type'])) == TradeType.BUY:
-            rate = amount_sold / amount_bought
-            amount = amount_bought
+        base_asset, fee, fee_currency, location, timestamp = process_rotki_generic_import_csv_fields(csv_row, 'Base Currency')  # noqa: E501
+        quote_asset = symbol_to_asset_or_token(csv_row['Quote Currency'])
+
+        # For a buy: spend quote asset, receive base asset
+        # For a sell: spend base asset, receive quote asset
+        if csv_row['Type'].strip().lower() == 'buy':
+            spend_asset = quote_asset
+            spend_amount = amount_sold
+            receive_asset = base_asset
+            receive_amount = amount_bought
         else:  # sell
-            rate = amount_bought / amount_sold
-            amount = amount_sold
-        trade = Trade(
-            timestamp=ts_ms_to_sec(timestamp),
-            location=location,
-            fee=fee,
-            fee_currency=fee_currency,
-            rate=Price(rate),
-            base_asset=asset,
-            quote_asset=symbol_to_asset_or_token(csv_row['Quote Currency']),
-            trade_type=trade_type,
-            amount=amount,
-            notes=csv_row['Description'],
+            spend_asset = base_asset
+            spend_amount = amount_sold
+            receive_asset = quote_asset
+            receive_amount = amount_bought
```

Comment:
spend amount and receive amount are the same in both cases so you don't need to duplicate the logic